### ISSUE TITLE
Hide / show tooltip is reversed sometimes when hovering over multiple tooltips

### ISSUE THREAD

**Issue by alvinvogelzang** [sir-dunxalot/ember-tooltips#25]
When hovering immediately from one to another tooltip the show / hide functionality is not working properly. Sometimes the hide / show is reversed, so hovering over a tooltip hides the tooltip and when I'm not hovering it shows the tooltip. 

Our application runs on Ember 1.13.10 and I tried ember-tooltips 0.5.2 and 0.4.0 but I'm seeing this behaviour in both versions. It doesn't matter if I use the tooltips as data-* attributes or as components / tooltip-on-parent component. 

Looks like the addon is hiding the new tooltip when hovering immediately from one tooltip to another and the `hide()` method is triggered after the `show()` method.

**Comment by alvinvogelzang** [comment]
It's also happening when showing / hiding / showing on the same tooltip immediately

**Comment by sir-dunxalot** [comment]
Thanks for the issue and your experimentation to narrow it down.

I think I have a good idea of what is causing this. I'll look into it and will report back.

**Comment by alvinvogelzang** [comment]
Great ;-)

**Comment by alvinvogelzang** [comment]
It looks like the bug occurs because of a setTimeOut in the darsain/tooltip vendor code. In `Tooltip.prototype.hide()` you have the following code: 

https://github.com/darsain/tooltip/blob/master/index.js#L354
```
// Remove the tip from the DOM when transition is done
	clearTimeout(this.aIndex);
	this.aIndex = setTimeout(function () {
		self.aIndex = 0;
		body.removeChild(self.element);
		self.hidden = true;
	}, duration);
```

When I remove the clearTimeOut and setTimeOut the bug doesn't occur:

```
  this.aIndex = 0;
  body.removeChild(this.element);
  this.hidden = true;
```

Probably it's the difference how the javascript is executed when using setTimeOut (also with 0) and without using setTimeOut. Not sure if there's something we can do about it in this addon or if we should fix it in the [darsain/tooltip](https://github.com/darsain/tooltip) code

**Comment by sir-dunxalot** [comment]
This seems like an issue best fixed in the dependency but we could overwrite the `Tooltip.prototype.hide()` method. My worry is that could be out of date/incompatible pretty quickly but we're currently pulling in the library from the `vendor` directory, not a package manager so I can't imagine it being much worse!

The set timeout is likely playing the role of allowing the tooltip to animate out (e.g. if `tooltipEffect='hover'`) so isn't that going to break the animations?

**Comment by alvinvogelzang** [comment]
I agree about fixing it in the dependency. 

I think you mean something like `tooltipEffectClass='fade'`. But you're right, without the setTimeOut animating out isn't working anymore so it's not an option to remove the setTimeOut.

**Comment by joukevandermaas** [comment]
any news on this?

**Comment by sir-dunxalot** [comment]
I don't have any immediate updates, sorry. We've been playing around with various attempts to fix some of the library finickiness in a product and hopefully something will come of that.

**Comment by alvinvogelzang** [comment]
@joukevandermaas I didn't work on a fix anymore. For my app I forked this repo and removed the setTimeOut. For me that's acceptable but the fade out isn't working this way.

**Comment by sir-dunxalot** [comment]
This has been fixed with release `1.0.0`: https://github.com/sir-dunxalot/ember-tooltips/releases/tag/1.0.0
